### PR TITLE
RTC-S1 Stub Implementation

### PR DESCRIPTION
**Pull request by collinss-jpl**
Initial addition of the stub implementation for the RTC-S1 PGE. 

Note that due to the specifics of how this PGE/SAS works, I also had to add a partial implementation of the file-naming conventions to ensure the PGE does not clobber the multiple per-burst outputs from the SAS.

I also took the liberty of refactoring some common code out of the build/test CI scripts into util.sh to keep things maintainable as we add more and more of these scrips.

### CODE REVIEW

File: src/opera/test/pge/rtc_s1/test_rtc_s1_pge.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        os.system(f"touch {join(input_dir, 'ORBIT.EOF')}")
+
+        os.system(f"touch {join(input_dir, 'dem.tif')}")
+
+        os.chdir(self.working_dir.name)
+
+    def tearDown(self) -> None:
+        """Return to test directory"""
+        os.chdir(self.test_dir)
```
Comment (david-inglish) [inline src/opera/test/pge/rtc_s1/test_rtc_s1_pge.py:69]: This looks like it is always "None".  Is this desired behavior?

File: src/opera/test/util/test_logger.py [hunk 2]
```diff
@@ -74,7 +76,7 @@ def setUp(self) -> None:
 
     def tearDown(self) -> None:
         """Return to starting directory"""
```
Comment (david-inglish) [inline src/opera/test/util/test_logger.py:78]: This comment may need an update?

**Review by david-inglish — APPROVED** [review]
I've made some notes where I had questions, but they don't seem to be major issues being in the unit test code.

File: .ci/scripts/build_cslc_s1.sh [hunk 3]
```diff
@@ -1,39 +1,13 @@
...
-      shift
-      shift
-      ;;
-    *)
-      echo "Unknown argument $1, ignoring..."
-      shift
-      ;;
-  esac
-done
+parse_build_args "$@"
```
Comment (JimHofman) [inline .ci/scripts/build_cslc_s1.sh:10]: Not sure what magic is going on here, but it sure saves some lines of code.

File: .ci/scripts/util.sh [hunk 4]
```diff
@@ -2,6 +2,119 @@
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 
+parse_build_args()
```
Comment (JimHofman) [inline .ci/scripts/util.sh:5]: I see good idea to factor this out.

File: src/opera/test/data/test_rtc_s1_config.yaml [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+            DynamicAncillaryFilesGroup:
+                AncillaryFileMap:
+                    dem_file: rtc_s1_test/input_dir/dem.tif
+
+            ProductPathGroup:
+                OutputProductPath: rtc_s1_test/output_dir
+                ScratchPath: rtc_s1_test/scratch_dir
+
+            PrimaryExecutable:
+                ProductIdentifier: RTC_S1
```
Comment (JimHofman) [inline src/opera/test/data/test_rtc_s1_config.yaml:23]: Will we be adding ProductVersion here as in the other PGEs?

File: src/opera/test/data/test_rtc_s1_config.yaml [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+                ProgramPath: mkdir
+                ProgramOptions:
+                    - '-p rtc_s1_test/output_dir/t069_147170_iw1/;'
+                    - '/bin/echo hello world > rtc_s1_test/output_dir/t069_147170_iw1/rtc_product.nc;'
+                    - '/bin/echo RTC-S1 invoked with RunConfig'
+                ErrorCodeBase: 300000
+                SchemaPath: pge/rtc_s1/schema/rtc_s1_sas_schema.yaml
+                IsoTemplatePath:
```
Comment (JimHofman) [inline src/opera/test/data/test_rtc_s1_config.yaml:31]: Similiarly, will we have CompositeReleaseID (not sure if we want to add it before ticket 149 is merged but we should remember these additions if they apply.

**Review by JimHofman — APPROVED** [review]
We should remember that we will probably be added the changes for Product Version and Composite Release ID as described in ticket 149.

File: .ci/scripts/build_cslc_s1.sh (hunk 3 above)
Comment (collinss-jpl) [reply .ci/scripts/build_cslc_s1.sh:10]: `"$@"` is bash shorthand for "array of all the command line variables", which includes the name of the script as the first variable (at $0). Basically this is just forwarding all the command line options to the parsing function which has been refactored into util.sh.

File: src/opera/test/data/test_rtc_s1_config.yaml (hunk 5 above)
Comment (collinss-jpl) [reply src/opera/test/data/test_rtc_s1_config.yaml:23]: Yes, I will likely merge your branch first, then update this branch to incorporate the new fields

File: src/opera/test/pge/rtc_s1/test_rtc_s1_pge.py (hunk 1 above)
Comment (collinss-jpl) [reply src/opera/test/pge/rtc_s1/test_rtc_s1_pge.py:69]: `test_dir` is initialized to `None` at line 32, but `setUpClass` (which is called before any calls to `setUp` or `tearDown`) sets it to the location of the test directory in our repository.

**Pull request merged**